### PR TITLE
Converts crossword clue-preview to es6

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

Converts crossword clue-preview to es6

## What is the value of this and can you measure success?

More es6!

## Does this affect other platforms - Amp, Apps, etc?

No

## Screenshots

N/A

## Tested in CODE?

No

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/crosswords/anagram-helper/clue-preview.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            entry.solved = !!entry.value;
+
+            return this.props.hasShuffled
+                ? (entry.value && entry) || unsolved.shift()
+                : entry;
+        });
+    },
+
+    // Checks a object in the form{",":[4,7]}
+    checkIfLetterHasSeparator(locations: Object, letterIndex: number): string {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/clue-preview.js]: Could probably type up the `locations` object

```
type locations {
  ,?: Array<number>
  -?: Array<number>
}
```

Syntax might need checking!

File: static/src/javascripts/projects/common/modules/crosswords/anagram-helper/main.js [hunk 2]
```diff
@@ -100,9 +100,13 @@ const AnagramHelper = React.createClass({
...
-        const entries = cells.map(
-            coords => this.props.grid[coords.x][coords.y]
-        );
+        const entries = cells.map(coords => {
+            const entry = this.props.grid[coords.x][coords.y];
+
+            entry.key = `${coords.x},${coords.y}`;
+
+            return entry;
+        });
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/main.js]: I wonder if we can one-line this badboy?

```
const entries = cells.map(coords => Object.assign({}, this.props.grid[coords.x][coords.y], { key: `${coords.x},${coords.y}` }));
```

💥 

Actually, perhaps this is indecipherable!

Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/main.js]: I like the suggestion! :sunglasses: 👍

**Review by SiAdcock — APPROVED** [review]
👏  congrats on your ES6 conversion completion

**Pull request merged**